lib/src/airship_flutter.dart

```diff
@@ -295,12 +293,20 @@ class Airship {
     return _eventStreams[eventType];
   }
 
-  /// Initializes Airship with an [appKey] and [appSecret].
+  /// Initializes Airship with an [AirshipConfig] or
+  /// [appKey] and [appSecret].
   ///
   /// Returns true if Airship has been initialized, otherwise returns false.
-  static Future<bool> takeOff(String appKey, String appSecret) async {
-    Map<String, String> args = {"app_key": appKey, "app_secret": appSecret};
-    return await _channel.invokeMethod('takeOff', args);
+  /// For backwards compatibility [config] can be appKey[String] or [AirshipConfig]
+  /// in next major release airship will migrate to [AirshipConfig]
+  static Future<bool> takeOff(final AirshipConfig config) async {
```

Comment:
do we need another method that accepts `String appKey, String appSecret`?

Reply:
No, this needs to be removed